src/tun/iface.go

```diff
@@ -1,42 +1,64 @@
 package tun
 
-const TUN_OFFSET_BYTES = 4
+const TUN_OFFSET_BYTES = 80 // sizeof(virtio_net_hdr)
+
+func (tun *TunAdapter) idealBatchSize() int {
+	return tun.iface.BatchSize()
```

Comment:
Minor point, but now that this is just calling `tun.iface.BatchSize()`, we could pretty easily remove this function and call that directly. Unless you think `idealBatchSize` helps with readability, I'm pretty in different to this, so I'm fine with leaving it in that case.

Reply:
Fixed.